### ISSUE TITLE
[BUG] Can't create session when the connection is closing

### ISSUE THREAD

**Issue by bebeo92** [Azure/azure-sdk-for-net#15514]
**Describe the bug**
While enqueue event into event hub, at the high load period times I received a log of error message via application insight and also weblog

```
System.InvalidOperationException. Details: Can't create session when the connection is closing..    at Microsoft.Azure.Amqp.AmqpConnection.AddSession(AmqpSession session, Nullable`1 channel)
   at Microsoft.Azure.Amqp.AmqpConnection.CreateSession(AmqpSessionSettings sessionSettings)
   at Microsoft.Azure.EventHubs.Amqp.AmqpEventDataSender.CreateLinkAsync(TimeSpan timeout)
   at Microsoft.Azure.Amqp.FaultTolerantAmqpObject`1.OnCreateAsync(TimeSpan timeout)
   at Microsoft.Azure.Amqp.Singleton`1.GetOrCreateAsync(TimeSpan timeout)
   at Microsoft.Azure.Amqp.Singleton`1.GetOrCreateAsync(TimeSpan timeout)
   at Microsoft.Azure.EventHubs.Amqp.AmqpEventDataSender.OnSendAsync(IEnumerable`1 eventDatas, String partitionKey)
   at Microsoft.Azure.EventHubs.Amqp.AmqpEventDataSender.OnSendAsync(IEnumerable`1 eventDatas, String partitionKey)
   at Microsoft.Azure.EventHubs.EventDataSender.SendAsync(IEnumerable`1 eventDatas, String partitionKey)
   at Microsoft.Azure.EventHubs.EventHubClient.SendAsync(IEnumerable`1 eventDatas, String partitionKey)
```

I do found another bug reported in here and try to update Microsoft.Azure.EventHubs from version 3.0.0 to version 4.3.0 and still see the issue occur
https://github.com/Azure/azure-sdk-for-net/issues/8761#issue-522435467

![94413116-d7b28f80-01a4-11eb-9f03-802a3a180336](https://user-images.githubusercontent.com/9253836/94508516-51e22300-023c-11eb-9c4d-21369f7111bd.png)

**Comment by bebeo92** [comment]
In addition, currently, we already have a retry logic for  `EventHubsException` not sure that we should add retry logic for this `InvalidOperationException` exception
This could be a critical issue for us in case this exception cause lose events

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team best able to assist.

**Comment by serkantkaraca** [comment]
Exception doesn't cause message loss. Does it recover if you retry?

**Comment by serkantkaraca** [comment]
This issue is fixed in release 4.3.1 below.

https://www.nuget.org/packages/Microsoft.Azure.EventHubs/4.3.1
